server/src/test/java/io/seata/server/lock/LockManagerTest.java

```diff
@@ -62,6 +66,84 @@ public void acquireLock_failed(BranchSession branchSession1, BranchSession branc
+     * @param branchSession2 the branch session 2
+     * @throws Exception the exception
+     */
+    @ParameterizedTest
+    @MethodSource("deadlockBranchSessionsProvider")
+    public void deadlockTest(BranchSession branchSession1, BranchSession branchSession2) throws Exception {
+        LockManager lockManager = new MemoryLockManagerForTest();
+        try {
+            CountDownLatch countDownLatch = new CountDownLatch(2);
+            new Thread(() -> {
+                try {
+                    lockManager.acquireLock(branchSession1);
+                } catch (TransactionException e) {
+                    e.printStackTrace();
+                } finally {
+                    countDownLatch.countDown();
+                }
+            }).start();
+            new Thread(() -> {
+                try {
+                    lockManager.acquireLock(branchSession2);
+                } catch (TransactionException e) {
+                    e.printStackTrace();
+                } finally {
+                    countDownLatch.countDown();
+                }
+            }).start();
+            // Assume execute more than 5 seconds means deadlock happened.
+            Assertions.assertTrue(countDownLatch.await(5, TimeUnit.SECONDS));
+        } finally {
+            lockManager.releaseLock(branchSession1);
+            lockManager.releaseLock(branchSession2);
+        }
+
+    }
+
+    /**
+     * Make sure two concurrent branchSession register process with different row key list, at least one process could success.
+     *
+     * @param branchSession1 the branch session 1
+     * @param branchSession2 the branch session 2
+     * @throws Exception the exception
+     */
+    @ParameterizedTest
+    @MethodSource("deadlockBranchSessionsProvider")
+    public void concurrentUseAbilityTest(BranchSession branchSession1, BranchSession branchSession2)  throws Exception {
+        LockManager lockManager = new MemoryLockManagerForTest();
+        final AtomicBoolean first = new AtomicBoolean();
+        final AtomicBoolean second = new AtomicBoolean();
+        CountDownLatch countDownLatch = new CountDownLatch(2);
+        new Thread(() -> {
+            try {
+                first.set(lockManager.acquireLock(branchSession1));
+            } catch (TransactionException e) {
+                e.printStackTrace();
+            } finally {
+                countDownLatch.countDown();
+            }
+        }).start();
+        new Thread(() -> {
+            try {
+                second.set(lockManager.acquireLock(branchSession2));
+            } catch (TransactionException e) {
+                e.printStackTrace();
+            } finally {
+                countDownLatch.countDown();
+            }
+        }).start();
+        // Assume execute more than 5 seconds means deadlock happened.
+        if (countDownLatch.await(5, TimeUnit.SECONDS)) {
+            Assertions.assertTrue(first.get() || second.get());
+        }
```

Comment:
Should the assertion be (first && !second) || (!first && second)?

Reply:
Based on single responsibility principle, first || second is enough.
I think `LockManagerTest#acquireLock_failed` already do what you want to do.

Reply:
If you said so, you lost some test cases, this test case concurrentUseAbilityTest only test `at least one process could success in concurrent case`, but `LockManagerTest#acquireLock_failed` is not responsible for concurrent use case. I think your assertion is not enough. What do you think?

Reply:
I think no harm done to make this assertion completion, it has nothing to do with `responsibility principle`.

Reply:
OK, i understand.

Reply:
Based on the latest code, don't you think if first and second are both fail, this test case should be fail?

Reply:
This only happen on some guy acquires table locks（tb_2，t:1,2,3,4,5）and doesn't release them. 
At now, both methods which use these table locks will release them.

Reply:
So, should we change the assertion `Assertions.assertTrue((first.get() && !second.get()) || (second.get() || !first.get()))` now?

Reply:
Done a hour ago.

Reply:
I think it should be `(first.get() && !second.get()) || (second.get() && !first.get())`. When second and first are both false, the test case should be fail.

Reply:
Sorry, i miswrite.

Reply:
No problem. Thanks.